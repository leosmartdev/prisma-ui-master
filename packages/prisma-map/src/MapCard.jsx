/*
 * COPYRIGHT Â© 2018 OROLIA GROUP AND/OR ITS AFFILIATES. ALL RIGHTS ARE STRICTLY RESERVED.
 * THIS SOURCE CODE CONTAINS PROPRIETARY AND CONFIDENTIAL INFORMATION AND DATA AND IS THE SOLE
 * PROPERTY OF OROLIA GROUP AND/OR ITS AFFILIATES.
 * THIS SOURCE CODE MUST NOT BE USED, DISSEMINATED, OR DISTRIBUTED EXCEPT FOR THE AGREED PURPOSE.
 * UNAUTHORIZED USE, REPRODUCTION, OR ISSUE TO ANY THIRD PARTY IS NOT PERMITTED WITHOUT THE PRIOR
 * WRITTEN CONSENT OF THE OROLIA GROUP.
 * THIS SOURCE CODE IS TO BE RETURNED TO THE OROLIA GROUP WHEN THE AGREED PURPOSE IS FULFILLED.
 * -------------------------------------
 */
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import PrismaPropTypes from './prisma-prop-types';

MapCard.propTypes = {
  /**
   * className This component passes the classname into the underlying <Card> component which
   * knows how to handle the className property.
   */
  className: PropTypes.string,
  /**
   * Anchor location of the map card over the map.
   * Valid values are:
   *
   * ----------------------------------------------
   * | top-left        top-center       top-right |
   * |                                            |
   * | middle-left   middle-center   middle-right |
   * |                                            |
   * | bottom-left   bottom-center   bottom-right |
   * ---------------------------------------------|
   */
  anchor: PropTypes.oneOf([
    'top-left',
    'top-center',
    'top-right',
    'middle-left',
    'middle-center',
    'middle-right',
    'bottom-left',
    'bottom-center',
    'bottom-right',
  ]),
  /**
   * Child elements to place inside the card. This is how your custom content should be sent into
   * the component.
   */
  children: PrismaPropTypes.children,
  /** @private withStyles */
  classes: PropTypes.shape({
    root: PropTypes.string,
  }).isRequired,
};

MapCard.defaultProps = {
  anchor: 'top-right',
  children: null,
  className: null,
};

function MapCard({
  classes,
  children,
  className,
  anchor,
}) {
  const classNames = [classes[anchor]];
  if (className) {
    classNames.push(className);
  }

  return (
    <Card
      classes={{ root: classes.root }}
      className={classNames.join(' ')}
      raised
    >
      {children}
    </Card>
  );
}

export default withStyles(theme => ({
  root: {
    pointerEvents: 'auto',
    margin: theme.spacing.unit * 2,
    display: 'inline-block',
    maxHeight: `calc(100% - ${theme.spacing.unit * 4}px)`,
    maxWidth: `calc(100% - ${theme.spacing.unit * 4}px)`,
    position: 'absolute',
    boxSizing: 'border-box',
  },
  'top-left': {
    top: 0,
    left: 0,
  },
  'top-center': {
    left: '50%',
    top: 0,
    marginLeft: 0,
    transform: 'translateX(-50%)',
  },
  'top-right': {
    top: 0,
    right: 0,
  },
  'middle-left': {
    left: 0,
    top: '50%',
    marginTop: 0,
    transform: 'translateY(-50%)',
  },
  'middle-center': {
    left: '50%',
    top: '50%',
    marginLeft: 0,
    marginTop: 0,
    transform: 'translate(-50%, -50%)',
  },
  'middle-right': {
    top: '50%',
    right: 0,
    marginTop: 0,
    transform: 'translateY(-50%)',
  },
  'bottom-left': {
    left: 0,
    bottom: 0,
  },
  'bottom-center': {
    left: '50%',
    bottom: 0,
    marginLeft: 0,
    transform: 'translateX(-50%)',
  },
  'bottom-right': {
    bottom: 0,
    right: 0,
  },
}))(MapCard);
