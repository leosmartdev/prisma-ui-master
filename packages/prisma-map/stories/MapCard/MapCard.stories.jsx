import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@storybook/react';
import { withStyles } from '@material-ui/core/styles';
import { select } from '@storybook/addon-knobs';

import Button from '@material-ui/core/Button';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';
import Map from '../../src/Map';
import MapCard from '../../src/MapCard';

const mapStories = storiesOf('Components/MapCard', module);

mapStories.add(
  'Basic Component',
  () => (
    <Map longitude={-90} latitude={1}>
      <MapCard>
        <CardHeader title="Header" subheader="Subheader" />
        <CardContent>
          <Typography variant="body1">This is some content in the card.</Typography>
        </CardContent>
        <CardActions>
          <Button size="small" onClick={() => console.info('Hello Click')}>
            Click Me
          </Button>
        </CardActions>
      </MapCard>
    </Map>
  ),
  { info: { inline: true } },
);

const anchorLocations = {
  'top-left': 'top-left',
  'top-center': 'top-center',
  'top-right': 'top-right',
  'middle-left': 'middle-left',
  'middle-center': 'middle-center',
  'middle-right': 'middle-right',
  'bottom-left': 'bottom-left',
  'bottom-center': 'bottom-center',
  'bottom-right': 'bottom-right',
};

mapStories.add('Map Card Location', () => (
  <Map>
    <MapCard anchor={select('anchor', anchorLocations, 'top-left')}>
      <CardHeader title="Header" subheader="Subheader" />
    </MapCard>
  </Map>
));

ChangeDimensionComponent.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string,
  }).isRequired,
};

function ChangeDimensionComponent({ classes }) {
  return (
    <Map>
      <MapCard className={classes.root}>
        <CardHeader title="Header" subheader="Subheader" />
        <CardContent>
          <Typography variant="body1">This is some content in the card.</Typography>
        </CardContent>
        <CardActions>
          <Button size="small" onClick={() => console.info('Hello Click')}>
            Click Me
          </Button>
        </CardActions>
      </MapCard>
    </Map>
  );
}

const Wrapped = withStyles({
  root: {
    width: '600px',
    height: '400px',
  },
})(ChangeDimensionComponent);

mapStories.add('Change dimensions', () => <Wrapped />);

const OverflowWrapped = withStyles({
  root: {
    width: '900px',
    height: '600px',
  },
})(ChangeDimensionComponent);

mapStories.add('Dimensions larger than the map are prevented', () => <OverflowWrapped />);
