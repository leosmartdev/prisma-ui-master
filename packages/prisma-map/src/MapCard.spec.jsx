import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MapCard from './MapCard';

describe('<MapCard>', () => {
  it('Renders', () => {
    const wrapper = shallow(<MapCard />);
    expect(wrapper).not.toBeNull();
  });

  it('snapshots', () => {
    const tree = renderer.create(<MapCard />);
    expect(tree).toMatchSnapshot();
  });

  describe('passes className through to <Card/>', () => {
    it('renders with correct dimensions when passed as props', () => {
      const wrapper = shallow(<MapCard className="foo" />);
      expect(wrapper.prop('className')).toEqual('foo'); // <MapCard>
      expect(wrapper.dive().prop('classes').root).toEqual(wrapper.prop('classes').root); // <Card>
      expect(wrapper.dive().prop('className')).toContain('foo'); // <Card>
    });
  });

  describe('anchor positions', () => {
    it('top-left', () => {
      const wrapper = shallow(<MapCard anchor="top-left" />);
      expect(wrapper.dive().prop('className')).toEqual('MapCard-top-left-2');
    });

    it('combines classNames and doesn\'t override user input className', () => {
      const wrapper = shallow(<MapCard anchor="top-left" className="foo" />);
      expect(wrapper.dive().prop('className')).toEqual('MapCard-top-left-2 foo');
    });
  });
});
