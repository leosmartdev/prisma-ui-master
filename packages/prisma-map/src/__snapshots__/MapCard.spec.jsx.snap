// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapCard> snapshots 1`] = `
<div
  className="MuiPaper-root-12 MuiPaper-elevation8-22 MuiPaper-rounded-13 MuiCard-root-11 MapCard-root-1 MapCard-top-right-4"
/>
`;
